### PR TITLE
feat(content-sidebar): initial Threaded Replies related changes

### PR DESCRIPTION
**Pull request by kkuliczkowski-box**
These are the initial changes for Threaded Replies.

* Option to disable features in `ContentSidebar`
* Added `hasReplies` flag to `ActivitySidebarProps` for future development

### CODE REVIEW

**Comment by jstoffan** [comment]
LGTM, but we'll need another reviewer since I authored part of the changes. @ConradJChan @tjuanitas?

File: src/elements/content-sidebar/ActivitySidebar.js
Comment (jstoffan) [reply src/elements/content-sidebar/ActivitySidebar.js:448]: I was trying to stay consistent with the `hasDetails`, `hasSkills`, etc. props that ContentSidebar exposes, since these could also be considered top-level options rather than options we want to downplay. That said, we could easily go with `isFeatureEnabled(features, 'activityFeed.versions.enabled');` if that's preferable.

Comment (ConradJChan) [reply src/elements/content-sidebar/ActivitySidebar.js:448]: Makes sense. I saw those as toggling the related sidebars in the `ContentSidebar` whereas the `features` affects what feed item types are available which is why it felt like a better fit. But either way is acceptable

File: src/elements/content-sidebar/__tests__/ActivitySidebar.test.js
Comment (tjuanitas) [inline src/elements/content-sidebar/__tests__/ActivitySidebar.test.js:363]: There are some more cases if we want to be exhaustive:
```
            annotationsEnabled | appActivityEnabled | versionsEnabled
            ${false}           | ${false}           | ${true}
            ${false}           | ${true}            | ${true}
            ${true}            | ${false}           | ${false}
            ${true}            | ${true}            | ${false}
```

File: src/elements/content-sidebar/ActivitySidebar.js
Comment (tjuanitas) [inline src/elements/content-sidebar/ActivitySidebar.js:721]: nit: `approvers` could be alphasorted

File: src/elements/content-sidebar/ActivitySidebar.js
Comment (tjuanitas) [inline src/elements/content-sidebar/ActivitySidebar.js:56]: Are we using this or is this for future changes?

**Review by tjuanitas — COMMENTED** [review]
Is this still WIP? Could we update the PR title?

File: src/elements/content-sidebar/ActivitySidebar.js
Comment (kkuliczkowski-box) [reply src/elements/content-sidebar/ActivitySidebar.js:458]: Yes, there is no tasks in Canvas.

File: src/elements/content-sidebar/ActivitySidebar.js
Comment (kkuliczkowski-box) [reply src/elements/content-sidebar/ActivitySidebar.js:56]: This is for future changes. More PRs will follow soon :)

File: src/elements/content-sidebar/ActivitySidebar.js
Comment (kkuliczkowski-box) [reply src/elements/content-sidebar/ActivitySidebar.js]: Done

File: src/elements/content-sidebar/__tests__/ActivitySidebar.test.js
Comment (kkuliczkowski-box) [reply src/elements/content-sidebar/__tests__/ActivitySidebar.test.js:363]: Technically those features do not depend on each other, but I've added missing combinations.

File: src/elements/content-sidebar/Sidebar.js
Comment (jstoffan) [reply src/elements/content-sidebar/Sidebar.js:248]: @kkuliczkowski-box, we can probably just conditionally show or hide the entire sidebar via CSS or the DOM using external state in Canvas.

**Review by tjuanitas — APPROVED** [review]
looks good to me

File: src/elements/content-sidebar/ActivitySidebar.js
Comment (ConradJChan) [reply src/elements/content-sidebar/ActivitySidebar.js:458]: Will the ability to disable the GET tasks call be added later?

File: src/elements/content-sidebar/__tests__/Sidebar.test.js
Comment (ConradJChan) [inline src/elements/content-sidebar/__tests__/Sidebar.test.js:213]: I've typically seen the usage of `wrapper.exists` vs `containsMatchingElement`. When should we use one versus the other?

Comment (kkuliczkowski-box) [reply src/elements/content-sidebar/__tests__/Sidebar.test.js]: I've changed to using `wrapper.exists` to keep the tests more consistent.

File: src/elements/content-sidebar/ActivitySidebar.js
Comment (kkuliczkowski-box) [reply src/elements/content-sidebar/ActivitySidebar.js]: I think this is the good moment to add that. I've updated the PR, GET request for fetching tasks will no longer take place when `hasTasks` is false.

File: src/api/__tests__/Feed.test.js
Comment (ConradJChan) [inline src/api/__tests__/Feed.test.js:539]: We've been trying to move away from using `toHave*` matchers and instead using `toBe*` matchers

**Review by ConradJChan — APPROVED** [review]

**Review by ConradJChan — APPROVED** [review]

**Pull request merged**
